test/groovy/com/sap/piper/ConfigurationMergerTest.groovy

```diff
@@ -63,6 +63,17 @@ class ConfigurationMergerTest {
         Assert.assertEquals(null, merged.veggie)
     }
 
+    @Test
+    void testMergeDeepStructureWithMissingDefaults(){
+        Map defaults = [others:[apples: 18]]
```

Comment:
I'm puzzled. Can somebody explain, why we do not have ```Map defaults = null```, please? I mean with that test here we do not provide the NPE in case of missing ```defaults = defaults ?: []```. Where is my mistake?

Reply:
If there is no default value, in the recursive call, defaults[key] will be null.
That was the issue I had.

Reply:
understood